### PR TITLE
ignore '.', '_' when filtering completion results

### PR DESCRIPTION
**Pull request by kevinushey**
This is a much simpler version of the previous PR. I realized we can accomplish our goal simply by munging the query token; ie, we don't need all of the special logic attached to `isSubsequence` and friends.

### CODE REVIEW

**Comment by kevinushey** [comment]
Good catch -- you're right that they'd be scored identically (which is not desired here).

I think we want the candidate set to be built on top of the 'fuzzy' token (with `.` and `_` stripped), but the ordering / score to be done with the non-stripped token. Let me play with this a bit more.

**Comment by kevinushey** [comment]
Okay, I think this is good to go now. `scoreMatch()` now accepts queries with characters not within the suggestion string, and penalizes those queries accordingly.

File: .gitignore [hunk 1]
```diff
@@ -30,3 +30,4 @@ CMakeLists.txt.user
 *swp
 .orig
 .vagrant
+.*.Rnb.cached
```
Comment (jmcphers) [inline .gitignore]: Did you mean to include this in the changelist? ATM it's added automatically by RStudio but it's going away soon (@jjallaire and I decided that we're going to bury these cache folders in `.Rproj.user`).

**Comment by jmcphers** [comment]
LGTM w/ changes. Thanks!

File: .gitignore (hunk 1 above)
Comment (kevinushey) [inline .gitignore]: Oops, sorry -- let me get rid of that change.

**Pull request merged**